hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonValueFunction.java

```diff
@@ -113,7 +117,8 @@ public T eval(final Row row, final ExpressionEvalContext context) {
 
         final JsonPath jsonPath;
         try {
-            jsonPath = pathCache.asMap().computeIfAbsent(path, JsonPathUtil::compile);
+            jsonPath = constantPathCache != null ? constantPathCache :
+                    pathCache.computeIfAbsent(path, JsonPathUtil::compile);
```

Comment:
We can extract the lambda to a static final field. Currently it creates a new instance for each call.

Reply:
Done